src/Sylius/Bundle/CoreBundle/Tests/Functional/StateMachine/OrderPaymentWorkflowTest.php

```diff
@@ -0,0 +1,182 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\Tests\Functional\StateMachine;
+
+use Sylius\Bundle\CoreBundle\StateMachine\StateMachineInterface;
+use Sylius\Component\Core\Model\Order;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+
+final class OrderPaymentWorkflowTest extends KernelTestCase
+{
+    /** @test */
+    public function it_applies_available_transition_for_order_payment_cart_status(): void
+    {
+        $stateMachine = $this->getStateMachine();
+        $order = new Order();
+
+        $stateMachine->apply($order, 'sylius_order_payment', 'request_payment');
+
+        $this->assertSame('awaiting_payment', $order->getPaymentState());
+    }
+
+    /**
+     * @test
+     *
+     * @dataProvider availableTransitionsForAwaitingPaymentState
+     */
+    public function it_applies_all_available_transitions_for_order_payment_awaiting_payment_state(
```

Comment:
What do you think using this technique from https://github.com/Sylius/Sylius/pull/15411/files to reduce code duplication?
![image](https://github.com/Sylius/Sylius/assets/870747/c2a81a91-0118-4832-b83c-c3c7e311fc18)

Reply:
This is a test, so for me readability is more important than code reduction, so I prefer separate tests for all states. But I understand your point, so we Will see what others say.

PS. If we decide to go with this solution, it would be a good idea to standardize it across all workflow tests